sally/models/stock_lot.py

```diff
@@ -0,0 +1,54 @@
+# -*- coding: utf-8 -*-
+
+from odoo import models, fields, api
+from odoo.exceptions import ValidationError
+
+
+class FlowerStockLot(models.Model):
+    _inherit = "stock.lot"
+
+    water_ids = fields.One2many("flower.water", "serial_id")
+    is_flower = fields.Boolean(
+        string='Is Flower Product?', related='product_id.is_flower')
+    needs_watering = fields.Boolean(
+        string='Needs Watering?', related='product_id.needs_watering')
+    user_ids = fields.Many2many('res.users', string='Gardner')
```

Comment:
you added the users on lot  instead of the production which is fine. however you are missing the domain in this case. 

Not a big deal